### PR TITLE
Detect cloud_provider properly on AWS large instances

### PR DESCRIPTION
**Pull request by ikapelyukhin**

### CODE REVIEW

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/272455/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp3. (Merged 98d0cf23bd1e70df8506848d2990b7c13963f521 into 65518a68ff5510b50185b16c9167dcf39a24736e)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/272457/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged d1288b03fdefacf397169f12bb59ba86f24ddb37 into 65518a68ff5510b50185b16c9167dcf39a24736e)

File: spec/connect/hwinfo/base_spec.rb [hunk 1]
```diff
@@ -39,10 +39,18 @@
     end
 
     context 'on AWS hypervisors' do
-      let(:dmidecode_output) { File.read(File.join(fixtures_dir, 'dmidecode_aws.txt')) }
+      context 'on regular instances' do
+        let(:dmidecode_output) { File.read(File.join(fixtures_dir, 'dmidecode_aws.txt')) }
+        it 'detects it' do
```
Comment (jmammen) [inline spec/connect/hwinfo/base_spec.rb]: Would be `it 'detects the provider'` a bit more descriptive?

File: spec/connect/hwinfo/base_spec.rb [hunk 2]
```diff
@@ -39,10 +39,18 @@
...
+          expect(subject.cloud_provider).to eq('Amazon')
+        end
+      end
 
-      it 'detects it' do
-        expect(subject.cloud_provider).to eq('Amazon')
+      context 'on large instances' do
+        let(:dmidecode_output) { File.read(File.join(fixtures_dir, 'dmidecode_aws_large.txt')) }
+        it 'detects it' do
```
Comment (jmammen) [inline spec/connect/hwinfo/base_spec.rb]: Same here.

File: spec/connect/hwinfo/base_spec.rb (hunk 1 above)
Comment (ikapelyukhin) [reply spec/connect/hwinfo/base_spec.rb]: Sure, done.

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/274378/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp3. (Merged 524158269ad2c12479ade5a58a81829f5e64c4c7 into 65518a68ff5510b50185b16c9167dcf39a24736e)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/274393/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 524158269ad2c12479ade5a58a81829f5e64c4c7 into 65518a68ff5510b50185b16c9167dcf39a24736e)

**Pull request merged**